### PR TITLE
Domain model and overall architecture for platform page with diagrams

### PR DESCRIPTION
**Pull request by zelima**

### CODE REVIEW

File: docs/developers/platform.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+## Domain model
+
+#### Data Package
+
+Each Data Package (DataSet) may have zero or more resources and one or more versions. Data Package is owned by Publisher.
```
Comment (anuveyatsu) [inline docs/developers/platform.md:5]: Not sure .. probably **one** or more resources

File: docs/developers/platform.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+#### Publisher
+
+**Publisher** is an organization which "owns" Data Packages. Publisher may have zero or more Data Packages. Publisher May also have one or more user.
+
+#### User
+
+User is an authenticated entity, that is member of Publisher organization.
+
+**Important:** Users to not have Data Packages, Publishers do. Users are *members* of Publishers.
```
Comment (anuveyatsu) [inline docs/developers/platform.md]: Users **do not** have Data Packages

File: docs/developers/platform.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  publisher --1..*--> user
+  publisher --0..*--> pkg
+end
+</div>
+
+## Overall Architecture
+
+Our platform architecture is built on several different components:
+
+* [CLI][dpm-py] - Command Line Interface for
```
Comment (anuveyatsu) [inline docs/developers/platform.md]: for ?

File: docs/developers/platform.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+* [Front-end Web Application][dpr-api] - API, Login & Sign-Up and Browse & Search
+* [Views and Renderer][dpr-js] - Graphs, Charts, Tables and other visualization tools
```
Comment (anuveyatsu) [inline docs/developers/platform.md]: I think it's unclear here, we need to rewrite

File: docs/developers/platform.md (hunk 1 above)
Comment (zelima) [reply docs/developers/platform.md:5]: Well, maybe it does not make sense but it is possible

**Pull request merged**